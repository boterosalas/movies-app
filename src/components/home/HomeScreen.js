import React, { useEffect, useState } from 'react';
import { BASE_URL } from '../../helpers/helpers';
import PosterGridMovies from '../shared/posterGridMovie/PosterGridMovies';
import SlideShow from '../shared/slideShow/SlideShow';

const HomeScreen = () => {

    const [movies, setMovies] = useState([]);
    const [moviesPosterGrid, setMoviesPosterGrid] = useState([]);
    const [page, setPage] = useState(1);

    const getMoviesInit = () => {
        fetch(`${BASE_URL}&page=1}`)
            .then(res => res.json())
            .then(({ results }) => {
                console.log(results)
                setMovies(p => [...p, ...results]);
                setMoviesPosterGrid(p => [...p, ...results]);
                setPage(p => p + 1);
            });
    }

    const getMoviesPosterGrid = () => {
        fetch(`${BASE_URL}&page=${page}`)
            .then(res => res.json())
            .then(({ results }) => {
                setMoviesPosterGrid(p => [...p, ...results]);
                setPage(p => p + 1);
            });
    }

    useEffect(() => {
        getMoviesInit();
    }, []);

    return (
        <>
            <header>
                <h1>En cartelera</h1>
            </header>
            <hr />
            {movies.length > 0 &&
                <>
                    <section className="row">
                        <div className="col">
                            <SlideShow movies={movies} />
                        </div>
                    </section>
                    <section className="mt-5">
                        <h2>Películas</h2>
                        <hr />
                        <PosterGridMovies moviesList={moviesPosterGrid} />
                    </section>
                    <div className="col-8 mx-auto mt-5 mb-5">
                        <button onClick={getMoviesPosterGrid} className="btn btn-primary btn-block">Cargar más</button>
                    </div>
                </>
            }
        </>
    )
}

export default HomeScreen
