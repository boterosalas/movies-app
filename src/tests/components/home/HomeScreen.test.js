import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import HomeScreen from '../../../components/home/HomeScreen';

describe('Test on <HomeScreen />', () => {

    test('Should display the component correctly', () => {
        const wrapper = shallow(<HomeScreen />);
        expect(wrapper).toMatchSnapshot();
    });


})