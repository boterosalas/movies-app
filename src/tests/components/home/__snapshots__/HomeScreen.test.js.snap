// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <HomeScreen /> Should display the component correctly 1`] = `
<Fragment>
  <header>
    <h1>
      En cartelera
    </h1>
  </header>
  <hr />
</Fragment>
`;
